### PR TITLE
add zboxcli's available-after params in share file test

### PR DESCRIPTION
**Pull request by Rizary**
Adding system_test for available-after in zboxcli (fix 0chain/blobber#439)

- blobber: https://github.com/0chain/blobber/pull/442
- gosdk: https://github.com/0chain/gosdk/pull/331
- zbox: https://github.com/0chain/zboxcli/pull/131
- system_test: https://github.com/0chain/system_test/pull/122

### CODE REVIEW

File: tests/cli_tests/zboxcli_share_file_test.go [hunk 1]
```diff
@@ -1437,6 +1437,178 @@ func TestShareFile(t *testing.T) {
...
+		}
+		output, err = shareFile(t, configPath, shareParams)
+		require.Nil(t, err, strings.Join(output, "\n"))
+		require.Len(t, output, 1, "share file - Unexpected output", strings.Join(output, "\n"))
+		require.Equal(t, "Cannot set available-after if the clientid is empty", output[0],
+			"share file - Unexpected output", strings.Join(output, "\n"))
+
+	})
+
+	t.Run("Download file before available-after params should fail ", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_share_file_test.go:1483]: Additional scenarios I think we need
download file during window should succeed 
download file after window should fail

File: tests/cli_tests/zboxcli_share_file_test.go [hunk 2]
```diff
@@ -1437,6 +1437,178 @@ func TestShareFile(t *testing.T) {
 			require.InEpsilon(t, expectedDownloadCostInZCN, diff, epsilon, "blobber [%v] read pool was deducted by [%v] rather than the expected [%v]", i, diff, expectedDownloadCostInZCN)
 		}
 	})
+
+	t.Run("Share to public with available-after params should fail ", func(t *testing.T) {
```
Comment (stewartie4) [inline tests/cli_tests/zboxcli_share_file_test.go:1441]: what does share to public mean here?
is share to public a feature that needs tested?

**Review by stewartie4 — CHANGES_REQUESTED** [review]
Thanks for the contribution, it's great to see new features being tested as they are written!
Some minor comments

Staging is broken right now so I ran tests manually using staging-556fedc2 image
image

File: tests/cli_tests/zboxcli_share_file_test.go (hunk 2 above)
Comment (Rizary) [reply tests/cli_tests/zboxcli_share_file_test.go:1441]: it's when you didn't pass clientID, you can't use `available-after` params

**Comment by Rizary** [comment]
> Thanks for the contribution, it's great to see new features being tested as they are written! Some minor comments
> 
> Staging is broken right now so I ran your new teat branch manually against the last working staging-556fedc2 image and your CLI feature branch image
> 
> It looks like your CLI if failing to build however https://github.com/0chain/system_test/runs/4389003394?check_suite_focus=true

hi thanks. Hmm, strange, I try to build locally and didn't get the error. How do we debug on this? I mean we use different version of gosdk (it's based on my PR on gosdk).

Another one is, the blobber should be changed, but I couldn't find it in the docker hub? what should I do? Can we just refer it to the branch like the one we did on zboxcli?

File: tests/cli_tests/zboxcli_share_file_test.go (hunk 2 above)
Comment (stewartie4) [reply tests/cli_tests/zboxcli_share_file_test.go:1441]: Understood
what i mean is i didn't realise that it's possible NOT to pass client id - i thought it was mandatory 
wondering if we need tests for that feature in general

**Comment by stewartie4** [comment]
@Rizary you need to publish the blobber docker image for your branch manually 
https://github.com/0chain/blobber/actions/workflows/build-&-publish-docker-image.yml
then pass that image into system tests as above

File: tests/cli_tests/zboxcli_share_file_test.go (hunk 2 above)
Comment (Rizary) [reply tests/cli_tests/zboxcli_share_file_test.go:1441]: I think we can add that test "not to pass client id".

File: tests/cli_tests/zboxcli_share_file_test.go (hunk 1 above)
Comment (Rizary) [reply tests/cli_tests/zboxcli_share_file_test.go:1483]: what do you mean by `during/after` window ?

**Comment by Rizary** [comment]
> @Rizary you need to publish the blobber docker image for your branch manually https://github.com/0chain/blobber/actions/workflows/build-&-publish-docker-image.yml then pass that image into system tests as above

Noted. I thought with GitHub action, we already publish every image whenever we create PR or push commit in that PR. I'll try to add the image manually into Docker Hub.

**Comment by cnlangzi** [comment]
> > @Rizary you need to publish the blobber docker image for your branch manually https://github.com/0chain/blobber/actions/workflows/build-&-publish-docker-image.yml then pass that image into system tests as above
> 
> Noted. I thought with GitHub action, we already publish every image whenever we create PR or push commit in that PR. I'll try to add the image manually into Docker Hub.

Yes. In blobber, docker image will be built and published with `staging:merge-[8-sha]` and `staging:merge`.  they can be used in system_test directly.

**Comment by Rizary** [comment]
> > > @Rizary you need to publish the blobber docker image for your branch manually https://github.com/0chain/blobber/actions/workflows/build-&-publish-docker-image.yml then pass that image into system tests as above
> > 
> > 
> > Noted. I thought with GitHub action, we already publish every image whenever we create PR or push commit in that PR. I'll try to add the image manually into Docker Hub.
> 
> Yes. In blobber, docker image will be built and published with `staging:merge-[8-sha]` and `staging:merge`. they can be used in system_test directly.

Thank you for confirming. Yes, I have used the `merge-sha` image in here: https://github.com/0chain/system_test/runs/4422822989?check_suite_focus=true

There are other failing tests which is not related with the feature I add. So I'm still investigate it.

**Comment by stewartie4** [comment]
yeah there are lots of panics being thrown from the CLI
image

File: tests/cli_tests/zboxcli_share_file_test.go (hunk 1 above)
Comment (stewartie4) [reply tests/cli_tests/zboxcli_share_file_test.go]: i mean the time window starting at the time of available after and ending at the time of expiry

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zboxcli-available-after  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | feature/available-after-param  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/available-after-param  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zboxcli-available-after  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | feature/available-after-param  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/available-after-param  |
| 0box  | feature/available-after-param  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zboxcli-available-after  |
| gosdk  | feature/available-after-param  |
| zboxcli  | feature/available-after-param  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | pr-442-24b649ee  |
| 0box  | feature/available-after-param  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by stewartie4** [comment]
@Rizary what's the status of this PR?

**Comment by Rizary** [comment]
I am still put it on hold due to another deadline. Will get back to it soon after I finish all of my task.

**Comment by stewartie4** [comment]
@princeparmar is your code being deployed to CI correctly? CI runs staging images unless you specify otherwise

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zboxcli-available-after  |
| gosdk  | feature/available-after-param  |
| zboxcli  | feature/zboxcli-available-after  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/zboxcli-available-after  |
| authorizer  | staging  |
| 0box  | feature/zboxcli-available-after  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by Rizary** [comment]
hi @stewartie4 , I'm working on it right now.

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zboxcli-available-after  |
| gosdk  | staging  |
| zboxcli  | feature/available-after-param  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | pr-442-76b882e5  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/zboxcli-available-after  |
| gosdk  | feature/available-after-param  |
| zboxcli  | feature/available-after-param  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feature/available-after-param  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by cnlangzi** [comment]
feature branch has been merged.

**Pull request closed without merging**